### DISCUSSION TITLE
When i use the GitHub prefers color scheme feature, i can't hide multiple languages

### DISCUSSION THREAD

**Discussion by xxxVitoxxx** [anuraghazra/github-readme-stats#3235] [Q&A] [answered]
Hello everyone,

### Use Markdown syntax
hiding multiple language is working.
![](https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&count_private=true&hide=html,javascript)

### Use img tag
hiding multiple language is working. 
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&count_private=true&hide=html,javascript">

### Use GitHub prefer color scheme feature
even if i hide multiple languages in query, it only hides one language. this is a bug?
<a>
  <picture>
    <source media="(prefers-color-scheme: dark)" 
      srcset="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&theme=radical&layout=compact&count_private=true&hide=html,javascript"
    />
    <source media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)" 
      srcset="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&count_private=true&hide=html,javascript"
    />
    <img align="center" width="410" src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&count_private=true&hide=html,javascript">
  </picture>
</a>

Thanks
Vito

**Comment by rickstaa** [comment]
Hey @xxxVitoxxx,

I think the problem you are experiencing is due to how GitHub handles caching of the card when creating dynamic themes using GitHub's [media feature](https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#use-githubs-new-media-feature). It looks like the card cache is not reloaded when parameters are changed. I tried to break the cache by appending `random=2341562`, but it still shows my previous entry. I will open a support ticket with github. I suggest you use another way to create a dynamic theme for now. I, for example, in my profile, use the [GitHub context](https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#use-githubs-theme-context-tag) method. I created a support ticket with GitHub to get this solved.

#### Regular card (MD)

![](https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact)

#### Languages hidden responsive theme (HTML)

<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&theme=dark&hide=go,CSS,javascript,HTML"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&hide=go,CSS,javascript,HTML"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&hide=go,CSS,javascript,HTML"/>
</picture>

```html
<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&theme=dark&hide=go,CSS,javascript,HTML"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&hide=go,CSS,javascript,HTML"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&hide=go,CSS,javascript,HTML"/>
</picture>
```

#### Hidden languages (HTML)

<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&hide=go,CSS,javascript,HTML" />

**Reply by xxxVitoxxx** [reply]
Thank you for answering the question! If possible, I still hope to use media feature. If GitHub has a solution, could you please inform me? I will be grateful to you.

**Reply by rickstaa** [reply]
Ofcourse! I will put the answer here when the GitHub support answered me 👍🏻.

**Reply by xxxVitoxxx** [reply]
thank you so much!

**Reply by rickstaa** [reply]
@xxxVitoxxx We got a response from GitHub support stating that they forwarded the issue to their engineering team.

![image](https://github.com/anuraghazra/github-readme-stats/assets/17570430/e7a6ad05-25f4-4888-a173-b778ecacee80)

**Reply by xxxVitoxxx** [reply]
thank you very much for your help!

**Reply by octatone** [reply]
👋 

Hi all, just wanted to clarify the root cause of the unexpected behavior here.

For `srcset` commas are a special character to delimit a collection of image urls, so they need to be properly escaped.  If you URI encode `&hide=go,CSS,javascript,HTML` to `&hide=go%2CCSS%2Cjavascript%2CHTML` it will just work. See [Mozilla srcset docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#srcset) for more information.

Additionally GitHub only supports one URL in `srcset` and drops any other comma delimited URLs. That is why if you check the rendered source, the canonical URL attribute ends with `&hide=go` instead of `&hide=go,CSS,javascript,HTML`; everything past the first unescaped comma will be dropped.

**Reply by rickstaa** [reply]
@octatone, excellent! Thanks a lot for explaining the issue. If you add your explanation with the correct syntax as an answer below, I will select it as the accepted answer ❤️‍🔥!

**Comment by qwerty541** [comment]
@xxxVitoxxx You can use [GitHub theme context tag](https://github.com/anuraghazra/github-readme-stats#use-githubs-theme-context-tag) to get desired result. 

<a href="https://github.com/xxxVitoxxx#gh-light-mode-only"><img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&count_private=true&hide=html,javascript" /></a>
<a href="https://github.com/xxxVitoxxx#gh-dark-mode-only"><img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&theme=radical&count_private=true&hide=html,javascript" /></a>

```md
<a href="https://github.com/xxxVitoxxx#gh-light-mode-only"><img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&theme=radical&count_private=true&hide=html,javascript" /></a>
<a href="https://github.com/xxxVitoxxx#gh-dark-mode-only"><img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&layout=compact&count_private=true&hide=html,javascript" /></a>
```

![image](https://github.com/anuraghazra/github-readme-stats/assets/53787217/4b56026a-6f9f-4011-be3d-647e42ff78eb)

**Reply by xxxVitoxxx** [reply]
Thank you for help 👍🏻.

**Comment by rickstaa** [comment, marked as answer]
I just received an email from GitHub support that validates the comment of @octatone.

![image](https://github.com/anuraghazra/github-readme-stats/assets/17570430/d0a9bc85-04fe-4725-9f07-ad4a16f4ff6d)

The right syntax should be:

```HTML
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&hide=go%2CCSS%2Cjavascript%2CHTML"/>
```

<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=xxxVitoxxx&hide=go%2CCSS%2Cjavascript%2CHTML"/>

**Comment by krosibahili** [comment]
Okkkk, let's goooo!
